text/0000-feature-metadata.md

```diff
@@ -0,0 +1,310 @@
+-   `public`: A boolean flag defaulting to `true` that indicates whether or not
+    downstream crates should be allowed to use this feature
+
+If a downstream crate attempts to use the features `baz` and `qux`, they will
+see messages like the following:
+
+```
+warning: feature `quux` on crate `mycrate` is deprecated since version
+  1.2.3: "don't use this!"
+
+error: feature `baz` on crate `mycrate` is private and cannot be used by
+  downstream crates
+```
+
+# Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+Validation and parsing of the new schema, described above, shoud be relatively
+straightforward. Each of the added keys is discussed individually in the
+following sections:
+
+## `requires`
+
+`requires` will take the place of the feature dependency array that currently
+exists. Semantics will remain unchanged.
+
+This is a required key. If there are no requirements, an empty list should be
+provided (`requires = []`).
+
+## `doc`
+
+`doc` is the most straightforward: it accepts markdown-flavored text, and should
+be thought of as the equivalent to a `#[doc(...)]` attribute. Like doc comments,
+the first line should be treated as a summary.
+
+There is nothing in this RFC that cargo `must` do with this action, since it is
+mainly intended for the consumption of `rustdoc` or `docs.rs`. However, it can
+be used for general diagnostic information such as during `cargo add` or
+a possible `cargo info` command.
+
+## `deprecated`
+
+`deprecated` should be thought of as the equivalent of the [`deprecated`]
+attribute in Rust source. The value can be a boolean, string, or an object with
+`since` or `note` keys. Schema rules are as follows:
+
+-   If a boolean value, `false` indicates not deprecated and `true` indicates
+    deprecated
+-   If an object, the keys `since` and/or `note` can be specified
+-   An empty object is not allowed to avoid the ambiguous
+    `foo = { deprecated = {} }`
+-   A string `foo = { deprecated = "my msg" }` will be equivalent to if that string
+    was specified in the `note` field:
+    `foo = { deprecated = { note = "my msg" } }`
+-   If not specified, the default is `false`
+
+If a downstream crate attempts to use a feature marked `deprecated`, Cargo
+should produce a warning. There are two exceptions to this:
+
+-   If a `since` is specified that is later than the crate's current version,
+    this warning should not be emitted.
+-   This warning should not be emitted for crates that reexport the feature
+    under a feature also marked deprecated. For example: crate `foo` exports
+    feature `phooey`, and crate `bar` exports feature `barred = ["foo/phooey"]`.
+    If `foo` markes `bar` as deprecated, checking `bar` will emit a warning
+    unless `barred` is also marked `deprecated.
+
+## `public`
+
+`public` is a boolean value that defaults to `true`. It can be thought of as
+`pub` in Rust source files, with the exception of being true by default. If set
```

Comment:
Just to confirm - are saying that `public = true` is a reasonable default here for consistency, but I should mention the reason for choosing this default?

Reply:
What I'm saying is `feature = {}` defaulting `public = true` is a reasonable default because it is consistent with what we'll do for `features = []` (and I don't think we should change `features = []`).  What we don't be consistent with is public/private dependencies and I think that is ok.

I think we should call out that we aren't being consistent with public/private dependencies (including a link to https://rust-lang.github.io/rfcs/1977-public-private-dependencies.html?) and that it is because the above reasons.

Reply:
I added a note about this in this section